quartz/components/scripts/checkbox.inline.ts

```diff
@@ -2,27 +2,31 @@ document.addEventListener("nav", () => {
   document.querySelectorAll(".task-list-item").forEach(addToggleHandler)
 })
 
-function addToggleHandler(listItem: Element) {
+function addToggleHandler(listItem: Element, index: number) {
   const toggleCheckbox = (e: any) => {
     const updatedCheckboxState: boolean = e.target.checked
-    const checkboxId = e.target.id
-    localStorage.setItem(`isCheckbox-${checkboxId}-enabled`, JSON.stringify(updatedCheckboxState))
+    const checkboxId = createCheckboxId(index)
+    localStorage.setItem(checkboxId, JSON.stringify(updatedCheckboxState))
   }
 
   const checkbox = extractCheckbox(listItem)
   if (!checkbox) return
-  handleCheckboxState(checkbox)
+  handleCheckboxState(checkbox, index)
   checkbox.addEventListener("change", toggleCheckbox)
 }
 
-function handleCheckboxState(checkbox: HTMLInputElement) {
-  const checkboxId = checkbox.id
-
-  const savedCheckboxState = localStorage.getItem(`isCheckbox-${checkboxId}-enabled`)
+function handleCheckboxState(checkbox: HTMLInputElement, index: number) {
+  const checkboxId = createCheckboxId(index)
+  const savedCheckboxState = localStorage.getItem(checkboxId)
   if (savedCheckboxState) checkbox.checked = JSON.parse(savedCheckboxState)
-  else localStorage.setItem(`isCheckbox-${checkboxId}-enabled`, JSON.stringify(checkbox.checked))
+  else localStorage.setItem(checkboxId, JSON.stringify(checkbox.checked))
 }
 
 function extractCheckbox(listItem: Element) {
   return listItem.querySelector('input[type="checkbox"]') as HTMLInputElement
```

Comment:
can we query on L2 for this directly instead of doing a nested query selector?

Reply:
Do you mean something like this?

```suggestion
  return listItem.getElementsByTagName('input')[0];
```